REFERENCE.md

```diff
@@ -0,0 +1,370 @@
+* [`resources_deep_merge`](#resources_deep_merge): Deeply merge a "defaults" hash into a "resources" hash like the ones expected by create_resources(). Internally calls the puppetlabs-stdlib f
+## Functions
+
+### cache_data
+Type: Ruby 3.x API
+
+Retrieves data from a cache file, or creates it with supplied data if the file doesn't exist
+
+Useful for having data that's randomly generated once on the master side (e.g. a password), but
+then stays the same on subsequent runs.
+
+Usage: cache_data(namespace, name, initial_data)
+Example: $password = cache_data('mysql', 'mysql_password', 'this_is_my_password')
+
+#### `cache_data()`
+
+Retrieves data from a cache file, or creates it with supplied data if the file doesn't exist
+
+Useful for having data that's randomly generated once on the master side (e.g. a password), but
+then stays the same on subsequent runs.
+
+Usage: cache_data(namespace, name, initial_data)
+Example: $password = cache_data('mysql', 'mysql_password', 'this_is_my_password')
+
+Returns: `Any`
+
+### default_content
+Type: Ruby 3.x API
+
+Takes an optional content and an optional template name and returns the
+contents of a file.
+
+Examples:
+
+    $config_file_content = default_content($file_content, $template_location)
+    file { '/tmp/x':
+      ensure  => 'file',
+      content => $config_file_content,
+    }
+
+#### `default_content()`
+
+Takes an optional content and an optional template name and returns the
+contents of a file.
+
+Examples:
+
+    $config_file_content = default_content($file_content, $template_location)
+    file { '/tmp/x':
+      ensure  => 'file',
+      content => $config_file_content,
+    }
+
+Returns: `Any`
+
+### dump_args
+Type: Ruby 3.x API
+
+dump_args - prints the args to STDOUT in Pretty JSON format.
+
+Useful for debugging purposes only. Ideally you would use this in
+conjunction with a rspec-puppet unit test.  Otherwise the output will
+be shown during a puppet run when verbose/debug options are enabled.
+
+#### `dump_args()`
+
+dump_args - prints the args to STDOUT in Pretty JSON format.
+
+Useful for debugging purposes only. Ideally you would use this in
+conjunction with a rspec-puppet unit test.  Otherwise the output will
+be shown during a puppet run when verbose/debug options are enabled.
+
+Returns: `Any`
+
+### echo
+Type: Ruby 3.x API
+
+This function output the variable content and its type to the
+debug log. It's similiar to the "notice" function but provides
+a better output format useful to trace variable types and values
+in the manifests.
+
+Example:
+$v1 = 'test'
+$v2 = ["1", "2", "3"]
+$v3 = {"a"=>"1", "b"=>"2"}
+$v4 = true
+# $v5 is not defined
+$v6 = { "b" => { "b" => [1,2,3], "c" => true, "d" => { 'x' => 'y' }}, 'x' => 'y', 'z' => [1,2,3,4,5,6]}
+$v7 = 12345
+
+echo($v1, 'My string')
+echo($v2, 'My array')
+echo($v3, 'My hash')
+echo($v4, 'My boolean')
+echo($v5, 'My undef')
+echo($v6, 'My structure')
+echo($v7) # no comment here
+
+# debug log output
```

Comment:
@alexjfisher so pumped to see REFERENCE.md! With these examples, you can either escape the hashes (to get rid of the h1 formatting) or use an `@example` tag to get noformat examples.

Reply:
Yep, the other functions need revisiting to fix up their output.  I was trying not to shave too many yaks this week! :)

I did notice one thing though.  Should I have expected my `@example` to appear in REFERENCE.md?